opacclient/libopac/src/main/java/de/geeksfactory/opacclient/apis/SLUB.kt

```diff
@@ -0,0 +1,571 @@
+            "Musical Score" to SearchResult.MediaType.SCORE_MUSIC,
+            "Website" to SearchResult.MediaType.URL,
+            "Journal, E-Journal" to SearchResult.MediaType.NEWSPAPER,
+            "Map" to SearchResult.MediaType.MAP,
+            "Audio" to SearchResult.MediaType.EAUDIO,
+            "Electronic Resource (Data Carrier)" to SearchResult.MediaType.EAUDIO,
+            "Image" to SearchResult.MediaType.ART,
+            "Microform" to SearchResult.MediaType.MICROFORM,
+            "Visual Media" to SearchResult.MediaType.ART
+    )
+
+    private val fieldCaptions = mapOf(
+            "format" to "Medientyp",
+            "title" to "Titel",
+            "contributor" to "Beteiligte",
+            "publisher" to "Erschienen",
+            "ispartof" to "Erschienen in",
+            "identifier" to "ISBN",
+            "language" to "Sprache",
+            "subject" to "Schlagwörter",
+            "description" to "Beschreibung",
+            "linksRelated" to "Info",
+            "linksAccess" to "Zugang",
+            "linksGeneral" to "Link"
+    )
+
+    private val ACTION_COPY = OpacApi.MultiStepResult.ACTION_USER + 1
+    private val pickupPoints = mapOf(
+            "a01" to "Zentralbibliothek Ebene 0 SB-Regal",
+            "a02" to "Zentralbibliothek, Ebene -1, IP Musik Mediathek",
+            "a03" to "Zentralbibliothek, Ebene -1, Lesesaal Sondersammlungen",
+            "a05" to "Zentralbibliothek, Ebene -2, Lesesaal Kartensammlung",
+            "a06" to "ZwB Rechtswissenschaft",
+            "a07" to "ZwB Erziehungswissenschaft",
+            "a08" to "ZwB Medizin",
+            "a09" to "ZwB Forstwissenschaft",
+            "a10" to "Bereichsbibliothek Drepunct",
+            "a13" to "Zentralbibliothek, Servicetheke",
+            "a14" to "Zentralbibliothek, Ebene -1, SB-Regal Zeitungen"
+    )
+
+    override fun init(library: Library, factory: HttpClientFactory) {
+        super.init(library, factory)
+        baseurl = library.data.getString("baseurl")
+        illRenewUrl = library.data.getString("illrenewurl")
+    }
+
+    override fun search(query: List<SearchQuery>): SearchRequestResult {
+        this.query = query
+        return searchGetPage(1)
+    }
+
+    override fun searchGetPage(page: Int): SearchRequestResult {
+        val queryUrlB = HttpUrl.get("$baseurl/?type=1369315142&tx_find_find[format]=data&tx_find_find[data-format]=app")
+                .newBuilder()
+                .addEncodedQueryParameter("tx_find_find[page]", page.toString())
+        for (sq in query) {
+            if (sq.value.isNotEmpty()) {
+                if (sq.searchField is DropdownSearchField) {  // access_facet
+                    queryUrlB.addEncodedQueryParameter("tx_find_find[facet][access_facet][${sq.value}]", "1")
+                } else {
+                    queryUrlB.addEncodedQueryParameter("tx_find_find[q][${sq.key}]", sq.value)
+                }
+            }
+        }
```

Comment:
I changed it on purpose from `addQueryParameter` to `addEncodedQueryParameter` in 0d52c8b5bb77b because of issues when the query string contains spaces (see second part of last item in commit message). Umlauts and &, /, # etc work OK.